### DISCUSSION TITLE
How to tell is renovate created a branch

### DISCUSSION THREAD

**Discussion by paul3535** [renovatebot/renovate#25037] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.4

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like to easily be able to determine in Renovate has created a branch (we create branches and merge to main). I need to be able to tell if a branch has been created, as well as if Renovate has merged a branch.

### What you tried so far.

I have tried parsing the renovate output, but that seems cumbersome.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by HonkingGoose** [comment]
> ### If you're self-hosting Renovate, tell us what version of Renovate you run.
> 
> 36.4

By the way, you're using an old version of Renovate. `37.6.1` is the latest version.

> I would like to easily be able to determine in Renovate has created a branch (we create branches and merge to main). I need to be able to tell if a branch has been created, as well as if Renovate has merged a branch.

Most people let Renovate create the MR and then review/merge it via the GitLab UI. Can you explain _why_ you're want to do something different?

**Reply by paul3535** [reply]
Sorry for the late response. We want to "break quickly" if we have any conflicting changes. Our main use is for keeping our internal go dependencies up to date, so we have any of those merged automatically if our pipelines pass. We have around 8 different projects, one of which has the main set of apis used throughout the projects and if there are any breaking changes that we didn't catch during review, we want to know right away. We have Renovate run every night and try and fix any new MRs (on failure it will create an MR) right away. We found when I first started up Renovate we had a few projects that hadn't been updated in months and it was a pain to get them up to date. Now we know the next day while the code is fresh in our minds.

As it turns out, I can parse the Renovate output and determine if a branch has been created and then can go based on that.

**Reply by HonkingGoose** [reply]
I may be missing something obvious here, but it still feels like you're making things harder for yourself somehow.

Did you see the [Renovate automerge docs, branch vs PR automerging](https://docs.renovatebot.com/key-concepts/automerge/#branch-vs-pr-automerging) section? I suspect it will be easier for you to let Renovate:

1. Create a branch with the update in it
2. Let GitLab (or your CI provider) run the tests
3. Automerge internal dependencies with passing tests
4. If tests fail, Renovate will open a PR

So instead of custom code, you rely on the signal: "Renovate opened a PR for a internal dependency due to failed tests".

Read the [Renovate docs, `automergeType` config option](https://docs.renovatebot.com/configuration-options/#automergetype) to learn more about branch automerging.

**Reply by paul3535** [reply]
My main concern was knowing if Renovate did indeed create a branch and merge it as we have a follow on build process that will build a release from the separate builds. Currently, at the end of a build a new release is kicked off incorporating all of the "latest" builds. When we run Renovate, we wanted to pause this follow on release build until all of the branches were merged so we didn't kick off 3 separate (and ultimately unnecessary as it takes ~1 hour to do this release build and the other builds would be finished before the release build) builds if 3 projects were updated.
I now disable the auto updating of the release build when Renovate starts and update the release project at the end with the updated builds and kick off the release build.

**Reply by HonkingGoose** [reply]
Checking if I understand you properly, does this sound right:

1. Making a release takes about an hour.
2. This means you want to put the most stuff into a single release build.
3. You pause the release build manually when there are Renovate branches.
4. After all Renovate branches are merged you start the release build by hand.

If that's the correct situation, you probably would like support for GitLab's merge train, where you only put the working stuff in a single commit. That single commit then triggers the release build:

- #5573

On the Renovate repository we use [Github's merge queue](https://docs.renovatebot.com/key-concepts/automerge/#github-merge-queue) to queue PRs, GitHub will make sure to only merge stuff that passes all the tests.
